node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+            return Err(FailedPayableDaoError::EmptyInput);
+        }
+
+        let hashes_vec: Vec<TxHash> = hashes.iter().cloned().collect();
+        let sql = format!(
+            "DELETE FROM failed_payable WHERE tx_hash IN ({})",
+            comma_joined_stringifiable(&hashes_vec, |hash| { format!("'{:?}'", hash) })
+        );
+
+        match self.conn.prepare(&sql).expect("Internal error").execute([]) {
+            Ok(deleted_rows) => {
+                if deleted_rows == hashes.len() {
+                    Ok(())
+                } else if deleted_rows == 0 {
+                    Err(FailedPayableDaoError::NoChange)
+                } else {
+                    Err(FailedPayableDaoError::PartialExecution(format!(
+                        "Only {} of the {} hashes has been deleted.",
+                        deleted_rows,
+                        hashes.len(),
+                    )))
+                }
+            }
+            Err(e) => Err(FailedPayableDaoError::SqlExecutionFailed(e.to_string())),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::accountant::db_access_objects::failed_payable_dao::FailureReason::{
+        NonceIssue, PendingTooLong,
+    };
+    use crate::accountant::db_access_objects::failed_payable_dao::FailureRetrieveCondition::ByReason;
+    use crate::accountant::db_access_objects::failed_payable_dao::{
+        FailedPayableDao, FailedPayableDaoError, FailedPayableDaoReal, FailureReason,
+        FailureRetrieveCondition,
+    };
+    use crate::accountant::db_access_objects::test_utils::FailedTxBuilder;
+    use crate::blockchain::test_utils::make_tx_hash;
+    use crate::database::db_initializer::{
+        DbInitializationConfig, DbInitializer, DbInitializerReal, DATABASE_FILE,
+    };
+    use crate::database::rusqlite_wrappers::ConnectionWrapperReal;
+    use crate::database::test_utils::ConnectionWrapperMock;
+    use masq_lib::test_utils::utils::ensure_node_home_directory_exists;
+    use rusqlite::{Connection, OpenFlags};
+    use std::collections::{HashMap, HashSet};
+    use std::str::FromStr;
+
+    #[test]
+    fn insert_new_records_works() {
+        let home_dir =
+            ensure_node_home_directory_exists("failed_payable_dao", "insert_new_records_works");
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let tx1 = FailedTxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = FailedTxBuilder::default()
+            .hash(make_tx_hash(2))
+            .reason(PendingTooLong)
+            .checked(1)
+            .build();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let txs = vec![tx1, tx2];
+
+        let result = subject.insert_new_records(&txs);
+
+        let retrieved_txs = subject.retrieve_txs(None);
+        assert_eq!(result, Ok(()));
+        assert_eq!(retrieved_txs.len(), 2);
```

Comment:
Redundant, given the next assertion which is stronger and bears the same information.